From 5866b2675eb4aba6ddc1e64d5afce60a4e8ccb0f Mon Sep 17 00:00:00 2001
From: Robert Marko <robimarko@gmail.com>
Date: Fri, 21 Aug 2020 15:26:00 +0200
Subject: [PATCH] arm64: dts: ipq8074: add SPMI arbiter node

IPQ8074 uses the SPMI arbiter to control the PM8074 PMIC connected
to it.
So lets add the SPMI node to enable the arbiter.

Signed-off-by: Robert Marko <robimarko@gmail.com>
---
 arch/arm64/boot/dts/qcom/ipq8074.dtsi | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

--- a/arch/arm64/boot/dts/qcom/ipq8074.dtsi
+++ b/arch/arm64/boot/dts/qcom/ipq8074.dtsi
@@ -529,5 +529,26 @@
 				      "axi_m_sticky";
 			status = "disabled";
 		};
+
+		spmi@200f000 {
+			compatible = "qcom,spmi-pmic-arb";
+			reg = <0x200f000 0x1000>,
+				<0x2400000 0x800000>,
+				<0x2c00000 0x800000>,
+				<0x3800000 0x200000>,
+				<0x200a000 0x000700>;
+			reg-names = "core", "chnls", "obsrvr", "intr", "cnfg";
+			interrupts = <GIC_SPI 190 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "periph_irq";
+
+			qcom,ee = <0>;
+			qcom,channel = <0>;
+
+			#address-cells = <2>;
+			#size-cells = <0>;
+
+			interrupt-controller;
+			#interrupt-cells = <4>;
+		};
 	};
 };
